src/components/ProbeEditor/ProbeEditor.tsx

```diff
@@ -1,344 +1,244 @@
-  let probe: Probe = TEMPLATE_PROBE;
-  if (id && probes) {
-    const idInt = parseInt(id, 10);
-    probe = probes.find((probe) => probe.id === idInt) ?? TEMPLATE_PROBE;
-  }
+import { getRoute } from 'components/Routing';
+import { SimpleMap } from 'components/SimpleMap';
+
+type ProbeEditorProps = {
+  actions?: ReactNode;
+  errorInfo?: { title: string; message: string };
+  onSubmit: (formValues: Probe) => void;
+  probe: Probe;
+  submitText: string;
+  supportingContent?: ReactNode;
+};
 
-  const formMethods = useForm<Probe>({ defaultValues: probe, mode: 'onChange' });
+export const ProbeEditor = ({
+  actions,
+  errorInfo,
+  onSubmit,
+  probe,
+  submitText,
+  supportingContent,
+}: ProbeEditorProps) => {
   const styles = useStyles2(getStyles);
+  const canEdit = canEditProbes(probe);
+  const form = useForm<Probe>({ defaultValues: probe, mode: 'onChange' });
+  const { latitude, longitude } = form.watch();
+  const handleSubmit = form.handleSubmit((formValues: Probe) => onSubmit(formValues));
+  const { errors, isSubmitting } = form.formState;
+  const alertRef = useRef<HTMLDivElement>(null);
+  const loading = form.formState.isSubmitting;
+
+  const getCoordsFromMap = useCallback(
+    ([long, lat]: number[]) => {
+      const significantDigits = 5;
+      form.setValue('longitude', +long.toFixed(significantDigits));
+      form.setValue('latitude', +lat.toFixed(significantDigits));
+      form.clearErrors(['longitude', 'latitude']);
+    },
+    [form]
+  );
 
-  const { execute: onSave, error } = useAsyncCallback(async (formValues: Probe) => {
-    reportEvent(FaroEvent.CREATE_PROBE);
-    // Form values always come back as a string, even for number inputs
-    formValues.latitude = Number(formValues.latitude);
-    formValues.longitude = Number(formValues.longitude);
-
-    if (!instance.api) {
-      throw new Error('Not connected to the Synthetic Montoring datasource');
-    }
+  const description = probe.public
+    ? 'Public probes are run by Grafana Labs and can be used by all users. They cannot be edited.'
+    : 'Private probes are operated by your organization and can only run your checks.';
 
-    if (probe.id) {
-      await instance.api.updateProbe({
-        ...probe,
-        ...formValues,
-      });
-      onReturn(true);
-    } else {
-      const info = await instance.api.addProbe({
-        ...probe,
-        ...formValues,
-        public: false,
-      });
-      setShowTokenModal(true);
-      setProbeToken(info.token);
+  useEffect(() => {
+    if (alertRef.current && errorInfo) {
+      alertRef.current?.scrollIntoView({ behavior: 'smooth' });
     }
-  });
-
-  const submissionError = error as unknown as SubmissionErrorWrapper;
-
-  if (error) {
-    reportError(error.message ?? error, FaroEvent.CREATE_PROBE);
-  }
+  }, [alertRef, errorInfo]);
 
-  if (!probe || !instance) {
-    return <div>Loading...</div>;
+  if (errors.labels) {
+    console.log(errors.labels);
```

Comment:
Did you still want this here? I've been wondering if these kinds of things could/should be shipped to Faro...

Reply:
Whoops, that was me getting annoyed trying to debug why some of the tests were failing and accidentally left it in.

Reply:
In regards to Faro. Maybe! I am still yet to play with it in-depth.

I think getting as much telemetry data as possible is worthwhile so we can make informed decisions when changing things.